programs/solana/programs/ics26-router/src/instructions/light_client_cpi.rs

```diff
@@ -98,5 +99,17 @@ pub fn verify_non_membership_cpi(
 
     invoke(&ix, &account_infos)?;
 
-    Ok(membership_msg.height)
+    // Get the return data from the light client
+    // Light client should return timestamp for non-membership verification
+    if let Some((_, return_data)) = get_return_data() {
+        if return_data.len() >= 8 {
+            let mut bytes = [0u8; 8];
+            bytes.copy_from_slice(&return_data[..8]);
+            return Ok(u64::from_le_bytes(bytes));
```

Comment:
This function probably doesn't need to return the height. If anything, it should return the timestamp on the counterparty chain at the time of the proof. This information is in the consensus state